client/src/pages/CreateGoal.tsx

```diff
@@ -0,0 +1,157 @@
+import { NumberOfDaysUntilDate } from '../utils';
+import SnailImage from '../imgs/snails/yellow-default.png'; // TODO: Change to utility function once other PR merged
+
+const GridWrapper = styled.div`
+  height: 85vh;
+  padding: 3vh;
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  align-items: center;
+  gap: 35px;
+`;
+
+const GoalCard = styled.div`
+  width: 50rem;
+  height: 60rem;
+  padding: 20px 15px;
+  background-color: ${COLORS.PURPLE_XTRALIGHT};
+  box-shadow: 10px 10px 10px #220d50;
+  border-radius: 15px;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const DeadlineWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  label {
+    white-space: nowrap; // Prevents text wrapping
+    font-weight: bold;
+  }
+  .datepicker {
+    font-size: 1.6rem;
+  }
+  * + * {
+    margin-inline-start: 4px;
+  }
+`;
+
+const SnailSection = styled.div`
+  width: 100%;
+  height: 19rem;
+  border-radius: 15px;
+  padding: 5px;
+  background-color: ${COLORS.PURPLE_LIGHT};
+
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-around;
+  align-items: center;
+`;
+
+const SnailSectionRightWrapper = styled.div`
+  width: 50%;
+  height: 85%;
+  text-align: center;
+
+  display: flex;
+  flex-flow: column wrap;
+  justify-content: space-around;
+  align-items: center;
+`;
+
+const H1 = styled.h1`
+  font-family: ${FONTS_MAIN};
+  font-style: italic;
+  font-weight: 600;
+  font-size: 2.4rem;
+  line-height: 2.9rem;
+  margin-bottom: 4px;
+`;
+
+function CreateGoal() {
+  const snailName = 'Snailosaurus'; // TODO: Get name from API
+  const [startDate, setStartDate] = useState(new Date());
+  const [numDays, setNumDays] = useState(0);
+  const numPages = 392; // TODO: Get page number from book info
+  return (
+    <PageWrapper pageTitle="Create a Goal">
+      <GridWrapper>
+        <LargeBookCard
+          bookTitle="This is the Title of a Book I could Write"
+          authorName="Joe Jonas"
+          bookCover=""
+          genres={[
+            'Horror',
+            'Mystery',
+            'Historical Fiction',
+            "Children's Literature",
+          ]}
+        />
+        <GoalCard>
+          <H1>Create a Goal</H1>
+          <DeadlineWrapper>
+            <Label htmlFor="deadline-datepicker">Goal Deadline:</Label>
+            <DatePicker
+              name="deadline-datepicker"
+              className="datepicker"
+              selected={startDate}
+              onChange={(newDate) => {
+                if (newDate) {
+                    const today = new Date();
+                    if (newDate < today) { // If the new date is in the past, it's invalid
+                        console.log("This date is in the past!");
+                    }
+                    else {
+                        setStartDate(newDate);
+                        setNumDays(NumberOfDaysUntilDate(newDate));
+                    }
```

Comment:
Logic for checking date @aiorgulescu

Reply:
Looks great. And then I think all we need to do after this is calculate the end date of the goal (startDate + numdays) and store that in local storage. Every time user opens up page or refreshes, we could pull that local storage value and check to see if current day - 1 = end date of goal. If it is, mark goal as failed. 

To store the goal dates in local storage, I think we can create a "goal" object that has the goal ID (should match the db ID) and the end date of the goal. I think that's all the info we need to store in local storage. 

Ex. [{goal_id: 1, end_date: 01/02/2023}, {goal_id: 2, end_date: 08/02/2023}]

Reply:
Instead of doing this, why don't we just compare the current date to the end date? We have to save the end date anyways and it's a quick check (similar to line 117).

I'm not against doing it the way you're talking about, just wondering the benefit!

Reply:
Wait is the`newDate` variable the end date? If so then yeah my suggestion is redundant my bad I misread